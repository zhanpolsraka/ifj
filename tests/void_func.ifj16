//0
class Main{
	static void run(){
		f1();
		f2();
		f3();
	}
	static void f1(){
		Data.x = 1;
	}
	static void f2(){
		Data.x = 2;
	}
}

class Data{
	static int x = 0;
}

class Outer{
	static void f3(){
		Data.x = 3;
	}
}