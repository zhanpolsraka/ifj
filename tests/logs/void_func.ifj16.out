---------------FILENAME: tests/void_func.ifj16---------------


Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> function with name "run"
Instruction stack item [22] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [21] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [20] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [19] -> end function
Instruction stack item [18] -> function with name "f1"
Instruction stack item [17] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [16] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [15] -> end function
Instruction stack item [14] -> function with name "f2"
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [11] -> end function
Instruction stack item [10] -> end class
Instruction stack item [9] -> class with name "Data"
Instruction stack item [8] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> end class
Instruction stack item [5] -> class with name "Outer"
Instruction stack item [4] -> function with name "f3"
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
Got out of Main
STEP 1 - non-func instructions
STEP 1 - non-func instructions
INSERT 0
---------------------------------------------------
N0: type is 22, values is 0
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 0 to 0x1337b70
---------------------------------------------------
N0: type is 22, values is 0
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSTR_CALL_FUNC
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
ASSIGN 1 to 0x1337b70
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSTR_CALL_FUNC
INSERT 2
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
ASSIGN 2 to 0x1337b70
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSTR_CALL_FUNC
INSERT 3
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
ASSIGN 3 to 0x1337b70
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Function -> run
	Function -> f1
	Function -> f2
Class Data
	Variable -> x, type INTEGER, value [3]
Class Outer
	Function -> f3


--------------INSTRUCTION LIST-------------

Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> function with name "run"
Instruction stack item [22] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [21] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [20] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [19] -> end function
Instruction stack item [18] -> function with name "f1"
Instruction stack item [17] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [16] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [15] -> end function
Instruction stack item [14] -> function with name "f2"
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [11] -> end function
Instruction stack item [10] -> end class
Instruction stack item [9] -> class with name "Data"
Instruction stack item [8] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> end class
Instruction stack item [5] -> class with name "Outer"
Instruction stack item [4] -> function with name "f3"
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



